libs/config/path_string.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// MustPathFromString is like NewPathFromString but panics on error.
+func MustPathFromString(input string) Path {
+	p, err := NewPathFromString(input)
+	if err != nil {
+		panic(err)
+	}
+	return p
+}
+
+// NewPathFromString parses a path from a string.
+//
+// The string must be a sequence of keys and indices separated by dots.
+// Indices must be enclosed in square brackets.
+// The string may include a leading dot.
+//
+// Examples:
+//   - foo.bar
+//   - foo[1].bar
+//   - foo.bar[1]
+//   - foo.bar[1][2]
+//   - .
+func NewPathFromString(input string) (Path, error) {
```

Comment:
I just realised that since this will be eventually used for working with real config, we might run into issues where the key itself can contain `.` like for example

```
 "spark_conf": {
          "spark.databricks.cluster.profile": "singleNode",
          "spark.master": "local[*, 4]"
        },
```

What if we format the path in the following way instead
```
"foo"[1]."bar"
```

Then if the key has spaces or dots the path will be less confusing, like

```
"foo.foo"[1]."bar baz"
```